### ISSUE TITLE
Property value `null` not caught by cfn-lint

### ISSUE THREAD

**Issue by Nr18** [awslabs/cfn-python-lint#246]
*cfn-lint version: (`cfn-lint --version`)* 
0.4.1

*Description of issue.*
The `AWS::Cognito::UserPool` resource will pass the cfn-lint tool but breaks in deployment if no lambda trigger function is given.

Please provide as much information as possible:
* Template linting issues:

```/Resources/UserPool/Type/LambdaConfig/PreSignUp] 'null' values are not allowed in templates (Service: AmazonCloudFormation; Status Code: 400; Error Code: ValidationError; Request ID: XXXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXXXX)```
  * Please provide a CloudFormation sample that generated the issue.

```cloudformation
  UserPool:
    Type: AWS::Cognito::UserPool
    Properties:
      LambdaConfig:
        PreSignUp: # No Value <-- this will resolve to null and accepted by cfn -lint
```

  * If present, please add links to the (official) documentation for clarification.
    * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-lambdaconfig.html

* Feature request:

I know not really handy to not fill in a value it was on my todo list but it passed the cfn-lint validation.
I guess that this scenario plays a role in more resources where you cannot have null as a value, in my opinion the linter should catch this if it's not allowed in CloudFormation.

**Comment by kddejong** [comment]
Interesting when I run that in mine I get the following results.  Your version should have had null checks in it so I was thinking that maybe the comments confused the parser.  However as you can see it still failes.  I tested across Python 2 and 3 and got same result so now I'm a little bit more confused.
```
E0000 Null value at line 7 column 18
test.yaml:8:19

➜  echo $?
2
➜  cat test.yaml
---
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  UserPool:
    Type: AWS::Cognito::UserPool
    Properties:
      LambdaConfig:
        PreSignUp: # No Value
```

**Comment by Nr18** [comment]
Hmpz, i will check it again maybe i made a stupid mistake or when removing custom stuff from the example it did validate.

I will also remove any custom rules that might cause a different behavior.. probably after the weekend

**Comment by Nr18** [comment]
Sorry my bad it was a stupid mistake... I trusted on my pipeline to do the validation and jumped to quick to conclusions thanks for the quick response!
